okhttp-tests/src/test/java/okhttp3/internal/DiskLruCacheTest.java

```diff
@@ -1536,6 +1536,78 @@ private void createNewCacheWithSize(int maxSize) throws IOException {
     assertNull(cache.get("a"));
   }
 
+  @Test public void noSizeCorruptionAfterCreatorDetached() throws Exception {
+    // Create an editor for k1. That editor's write privileges away by clearing the cache.
+    DiskLruCache.Editor editor = cache.edit("k1");
+    setString(editor, 0, "a");
+    setString(editor, 1, "a");
+    cache.evictAll();
+
+    // Create a new value in its place.
+    set("k1", "bb", "bb");
+
+    // Committing the detached editor should not change the cache's size.
+    editor.commit();
+    assertEquals(4, cache.size());
```

Comment:
Can we assert this before as well so that "should not change" is materialized in the test?